test/e2e/destroypod_test.go

```diff
@@ -286,10 +323,18 @@ func TestDestroyPodWithRequests(t *testing.T) {
 		time.Sleep(time.Second)
 	}
 
-	// And immeditately kill the pod.
-	podToDelete := pods.Items[0].Name
-	t.Logf("Deleting pod %q", podToDelete)
-	clients.KubeClient.Kube.CoreV1().Pods(test.ServingNamespace).Delete(podToDelete, &metav1.DeleteOptions{})
+	// And immeditately kill the pod or activators.
+	switch testCase {
+	case "pods":
+		podToDelete := pods.Items[0].Name
```

Comment:
The pod querying logic should go into here as well. Could we potentially also do a `DeleteCollection` in combination with a LabelSelector for the revision?